specs/dwritecore/DWriteCore.md

```diff
@@ -0,0 +1,255 @@
+# DWriteCore
+
+DWriteCore is the Project Reunion version of DirectWrite, the text rendering and layout
+component that ships in Windows. DWriteCore implements the same API as DirectWrite,
+with a few additions described in this document. See the [DirectWrite documentation](https://docs.microsoft.com/en-us/windows/win32/api/_directwrite/)
+for more information.
+
+For an application that already uses DirectWrite, switching to DWriteCore requires minimal changes:
+
+ - Add a reference to the Project Reunion package.
+ - Include dwrite_core.h instead of dwrite_3.h.
+ - Link to DWriteCore.lib instead of DWrite.lib.
+
+In return, the application gets the benefits of Project Reunion, namely, access to the newest APIs
+and functionality regardless of what version of Windows your customer is running.
+
+# Background
+
+<!-- TEMPLATE
+    Use this section to provide background context for the new API(s)
+    in this spec. 
+
+    This section and the appendix are the only sections that likely
+    do not get copied into any official documentation, they're just an aid
+    to reading this spec. 
+    
+    If you're modifying an existing API, included a link here to the
+    existing page(s) or spec documentation.
+
+    For example, this section is a place to explain why you're adding this
+    API rather than modifying an existing API.
+
+    If you're writing a "converged" API add links into docs.microsoft.com
+    for the existing Win32 or WinRT APIs that are being converged.
+-->
+
+<!-- TEMPLATE
+    For example, this is a place to provide a brief explanation of some dependent
+    area, just explanation enough to understand this new API, rather than telling
+    the reader "go read 100 pages of background information posted at ...". 
+-->
+
+The DWriteCore API is the same as the existing public DirectWrite API, with a few additions. These
+additions should eventually be incorporated into the in-box DirectWrite as well, so it is appropriate
+to think of the DWriteCore API as just the latest version of the DirectWrite API. This document focuses 
+primarily on the additions.
+
+## API Versioning
+
+The DirectWrite API has grown over time, and many APIs have multiple versions: IDWriteFactory, 
+IDWriteFactory1, IDWriteFactory2, etc. Prior to Windows 10, each new version got a new header file.
+Starting in TH1, new APIs have been added to dwrite_3.h, but conditionalized based on NTDDI_VERSION
+as follows:
+
+```c++
+#if NTDDI_VERSION >= NTDDI_WIN10_MN
```

Comment:
These versions aren't meaningful to developers; consider adding a "// Windows 10 release 1910" or whatever

Reply:
DISCUSSION: Ifdefs are carried from the public headers, which require versioning so apps can restrict API use. DWriteCore lets developers always use the APIs even if they're not on `_MN`.